qcodes/dataset/sqlite_base.py

```diff
@@ -354,16 +364,14 @@ def perform_db_upgrade(conn: SomeConnection, version: int=-1) -> None:
           'newest version'
     """
 
-    upgrade_actions = [perform_db_upgrade_0_to_1, perform_db_upgrade_1_to_2,
-                       perform_db_upgrade_2_to_3]
-    newest_version = len(upgrade_actions)
+    newest_version = len(_UPGRADE_ACTIONS)
```

Comment:
Shall this be extracted as well? well, at least it is extracted in #1389 in order to be used for testing.